templates/rhpam74-authoring-ha-datagrid.yaml

```diff
@@ -1181,11 +1182,16 @@ objects:
     labels:
       application: "${APPLICATION_NAME}"
       service: "${APPLICATION_NAME}-kieserver"
+      services.server.kie.org/kie-server-id: "${APPLICATION_NAME}-kieserver"
     annotations:
       template.alpha.openshift.io/wait-for-ready: "true"
   spec:
+    revisionHistoryLimit: 10
```

Comment:
What does this attribute mean, and how was the value of "10" determined? Does it need to be parameterized?

Reply:
The OpenShiftStartupStrategy is depending on DeploymentConfig rollout capability to propagate and synchronized configuration changes between KIE Server Pods. By default, DC keeps a history of previous ReplicationController (aka RC) (See below example, only last 10 historical records, rc-142 to rc-151 are kept) without any limitation of total number of records which will lead to a maintenance nightmare. Be having such parameter,  we believe it provides sufficient undo track records (aka 10) and mitigate the risk of etcd exhaustion. At the moment, we don't see much value for parameterizing it.  

```
NAME                                 REVISION   DESIRED   CURRENT   TRIGGERED BY
deploymentconfigs/myapp-kieserver    152        0         0         config,image(rhpam73-kieserver-openshift:1.0)
deploymentconfigs/myapp-rhpamcentr   55         0         0         config,image(rhpam73-businesscentral-openshift:1.0)

NAME                      HOST/PORT                                       PATH      SERVICES           PORT      TERMINATION   WILDCARD
routes/myapp-kieserver    myapp-kieserver-myproject73.127.0.0.1.nip.io              myapp-kieserver    <all>                   None
routes/myapp-rhpamcentr   myapp-rhpamcentr-myproject73.127.0.0.1.nip.io             myapp-rhpamcentr   http                    None

NAME                     DESIRED   CURRENT   READY     AGE
rc/myapp-kieserver-142   0         0         0         38d
rc/myapp-kieserver-143   0         0         0         38d
rc/myapp-kieserver-144   0         0         0         38d
rc/myapp-kieserver-145   0         0         0         37d
rc/myapp-kieserver-146   0         0         0         37d
rc/myapp-kieserver-147   0         0         0         30d
rc/myapp-kieserver-148   0         0         0         30d
rc/myapp-kieserver-149   0         0         0         25d
rc/myapp-kieserver-150   0         0         0         25d
rc/myapp-kieserver-151   0         0         0         25d
rc/myapp-kieserver-152   0         0         0         19d
```

Reply:
Got it - thank you for the explanation.